src/mqtt_client/mqtt_client.rs

```diff
@@ -253,17 +265,27 @@ where
             )
             .is_ok()
         {
-            self.state
-                .process_event(sm::Events::IndicatedAlive)
-                .unwrap();
+            self.state.process_event(sm::Events::IndicatedLife).unwrap();
         }
     }
 
-    /// Update the MQTT interface and service the network
+    /// Update the MQTT interface and service the network. Handle any settings changes.
+    ///
+    /// # Args
+    /// * `handler` - A closure called with updated settings that can be used to apply current
+    ///   settings or validate the configuration.
     ///
     /// # Returns
     /// True if the settings changed. False otherwise.
-    pub fn update(&mut self) -> Result<bool, minimq::Error<Stack::Error>> {
+    pub fn handled_update<F, E, T>(
```

Comment:
Ah. It took me five attempts to understand what "handled_update()" means.
Maybe just unify and have `update<F, E, T>(&mut self, handler: Option<F>)`?

Reply:
My onle goal with adding `handled_update()` was that it was a way to maintain backwards compatibility with the previous API.

The proposed API is really unergonomic if you don't have a handler, since Rust will complain that the type `F` cannot be automatically determined.

Reply:
Ack. Then maybe worth thinking a bit more about the docstring to make it clear what the function is and does. I think that would help me already.

Reply:
I added a complete code example to show the user.